### ISSUE TITLE
Features to speed up training

### ISSUE THREAD

**Issue by alejandroxag** [Nixtla/neuralforecast#5]
Recently, we found a couple of tricks to speed up the training processes:

- Data Parallelism: torch.nn.DataParallel(module) (https://pytorch.org/docs/stable/generated/torch.nn.DataParallel.html). This function automatically splits the input in the forward pass among the available GPUs and automatically collects and adds up the gradients in the backward pass.
- Automatic Mixed Precision: torch.cuda.amp (https://pytorch.org/docs/stable/generated/torch.nn.DataParallel.html). This package provides functions to automatically cast the datatype of the tensors to optimize the threshold between precision and computing speed.

**Comment by FedericoGarza** [comment]
With pytoch lightning we can now use those tricks.
